lib/modules/datasource/cpan/index.ts

```diff
@@ -0,0 +1,137 @@
+import { logger } from '../../../logger';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { getElapsedMinutes } from '../../../util/date';
+import { newlineRegex, regEx } from '../../../util/regex';
+import { copystr } from '../../../util/string';
+import * as perlVersioning from '../../versioning/perl';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+type Package = {
+  release: Release;
+  distribution: string;
+};
+
+let lastSync = new Date('2000-01-01');
+let packages: Record<string, Package> = Object.create(null); // Because we might need a "constructor" key
+let contentLength = 0;
+
+const pathExtensionPattern = regEx(/(\.tgz|\.tar\.gz)$/);
+const pathPattern = regEx(/^.+\/(.+)-(?:\d+(?:\.\d+)*)(?:\.tgz|\.tar\.gz)$/);
+
+// Note: use only for tests
+export function resetCache(): void {
+  lastSync = new Date('2000-01-01');
+  packages = Object.create(null);
+  contentLength = 0;
+}
+
+export class CpanDatasource extends Datasource {
+  static readonly id = 'cpan';
+
+  constructor() {
+    super(CpanDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = ['https://www.cpan.org'];
+
+  override readonly defaultVersioning = perlVersioning.id;
+
+  @cache({
+    namespace: `datasource-${CpanDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `${packageName}`,
+  })
+  override async getReleases({
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    await this.syncVersions();
+    const pkg = packages[packageName];
+    if (!pkg) {
+      return null;
+    }
+    const dep: ReleaseResult = {
+      releases: [pkg.release],
+      changelogUrl: `https://metacpan.org/dist/${pkg.distribution}/changes`,
+      homepage: `https://metacpan.org/pod/${packageName}`,
+    };
+    return dep;
+  }
+
+  async updateCpanVersions(): Promise<void> {
+    const url = 'https://www.cpan.org/modules/02packages.details.txt';
+    const options = {
+      headers: {
+        'accept-encoding': 'identity',
+        range: `bytes=${contentLength}-`,
+      },
+    };
+    let newLines: string;
+    try {
+      logger.debug('CPAN: Fetching www.cpan.org versions');
```

Comment:
```suggestion
      logger.trace({packageName, registryUrl}, 'Fetching versions');
```
IMO `trace` level is here and for similar messages enough.    
Please use variables instead of hard coded values as the user could overwrite these. e.g. with an internal registry